vllm/utils.py

```diff
@@ -517,3 +518,54 @@ def init_cached_hf_modules():
     """
     from transformers.dynamic_module_utils import init_hf_modules
     init_hf_modules()
+
+
+def nccl_integrity_check(filepath):
+    """
+    when the library is corrupted, we cannot catch
+    the exception in python. it will crash the process.
+    instead, we use the exit code of `ldd` to check
+    if the library is corrupted. if not, we will return
+    the version of the library.
+    """
+    exit_code = os.system(f"ldd {filepath} 2>&1 > /dev/null")
+    if exit_code != 0:
+        raise RuntimeError(f"Failed to load NCCL library from {filepath} .")
+    import ctypes
+
+    import torch  # noqa
```

Comment:
already imported

Reply:
thanks! fixed in [8a06dc3](https://github.com/vllm-project/vllm/pull/4155/commits/8a06dc3c0329b652d1364522988240e5930840cb).